src/classes/groundplaces.ts

```diff
@@ -33,27 +34,88 @@ export class GroundPlacesController {
 
   /**
    * @description Init GroundPlaces file.
-   * @param {string|undefined} groundPlacesFile - The file to manipulate, can only be JSON for now.
+   * @param {GroundPlacesFile|undefined} groundPlacesFile
+   * The file to manipulate, can only be JSON for now.
+   *
    * If you provide empty value, the default file will be the file retrieved from Amazon S3.
+   * @returns {void}
    */
-  public init(groundPlacesFile?: GroundPlaces): void {
+  public init(groundPlacesFile?: GroundPlacesFile): void {
     if (groundPlacesFile) {
       this.storageService.initFile(groundPlacesFile);
     } else {
-      const groundPlacesFileS3: GroundPlaces = this.webService.downloadDistantGroundPlacesMaster();
+      const groundPlacesFileS3: GroundPlacesFile = this.webService.downloadDistantGroundPlacesMaster();
 
       this.storageService.initFile(groundPlacesFileS3);
     }
   }
 
   /**
-   * @description Returns a list of places.
+   * @description Returns a list of ground places.
    * @param {string} query - Can be a name, a Gpuid, a unique name or other name.
-   * @param {AutoCompleteFilters} filters - Filters with different options (StopGroup, StopCluster, Serviced, SegmentProvider).
-   * @returns {AutoComplete[]}
+   * @param {AutoCompleteFilters[]|undefined} filters - Filters with different options (StopGroup, StopCluster, Serviced, SegmentProvider).
+   *
+   * If you do not give filters, the list will not be filtered.
+   * @returns {GroundPlace[]}
    */
   // @ts-ignore
-  public autocomplete(query: string, filters: AutoCompleteFilters[]): AutoComplete[] {}
+  public autocomplete(query: string, filters?: AutoCompleteFilters[]): GroundPlace[] {
+    const groundPlaces: GroundPlace[] = this.getGroundPlaces();
+
+    // Method toLowerCase() is used because the includes() method is case sensitive
+    const currentQuery: string = query.toLowerCase();
+
+    const isFilterByStopGroupActive: boolean = filters?.includes(AutoCompleteFilters.STOP_GROUP);
+    const isFilterByStopClusterActive: boolean = filters?.includes(AutoCompleteFilters.STOP_CLUSTER);
+    const isFilterByServicedActive: boolean = filters?.includes(AutoCompleteFilters.SERVICED);
+    const isFilterWithChildrenActive: boolean = filters?.includes(AutoCompleteFilters.SEGMENT_PROVIDER_STOP);
+
+    return groundPlaces.filter(
```

Comment:
Ths filter this really complex. 
I might be wrong but i think we can do simpler.

Reply:
ie : 
`if ( !autoCompleteFilters.includes(place.type as AutocompleteFilter ) return ;`

Reply:
first. and then serviced and then the different properties.

Reply:
dont really understand your point.
- merge stopgroup and stopcluster filters ?
- put stopgroup and stopcluster filters before the search by query ?

Reply:
Some improvements to the funcrtion :
- I will first check if filters are not provided, if it is the case I return the place directly if it has validated the parameters of the search.
- If there are filters, I filter with the filters by type of place and by serviced.
- And finally I filter with the segmentproviderStop inside childrens (which is the most expensive filter) at the end